### DISCUSSION TITLE
What kind of permissioned blockchain CCF is based on?

### DISCUSSION THREAD

**Discussion by liuchao11** [microsoft/CCF#5282] [Q&A]
There are some things I don't understand. Firstly, CCF is a permissioned blockchain. So, is CCF based on Hyperledger or other permissioned? Secondly, If I want to hold Hyperledger inside CCF. Is there any work I should do?  What is the best way to proceed?

**Comment by achamayou** [comment]
@liuchao11 CCF makes use of a ledger, but we think of it more as a framework to build confidential services than as a traditional permissioned blockchain. You can find an overview in our [documentation](https://microsoft.github.io/CCF/main/overview/what_is_ccf.html), but if anything is unclear, questions are always welcome.

CCF is not based on Hyperledger, or any other blockchain project. I am not sure how to hold Hyperledger inside CCF, nor quite how you would proceed, but I would suggest reading up on how to [build CCF applications](https://microsoft.github.io/CCF/main/build_apps/index.html) and checking if the functionality you need is present.

CCF applications are built in C++ or JavaScript and have access to a KV store that is replicated through a ledger. At least one HyperLedger project, PDO, is [making use of CCF](https://github.com/hyperledger-labs/private-data-objects/tree/bdbfd79c22dadc82fa5c6f275397966bde1063f5/ledgers/ccf). I am not sure if that's a useful pointer for your purposes though.
